src/session.ts

```diff
@@ -3,15 +3,45 @@ import Assignment from './assignment';
 import AssignmentOverride from './assignmentOverride';
 import TestTrackConfig from './testTrackConfig';
 import Visitor from './visitor';
+import MixPanelAnalytics from './mixpanelAnalytics';
 
-let loaded = null;
+let loaded: null | ((value?: Visitor | PromiseLike<Visitor>) => void) = null;
```

Comment:
for my understanding, is this to support how promises work? you can either pass in the actual resolved object or another promise?

Reply:
Yes, the type signature for `resolve` in the Promise constructor is:

```ts
declare type PromiseConstructorLike = new <T>(executor: (resolve: (value?: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void) => PromiseLike<T>;
```

Where `T` stands for our `Visitor` interface.